kernel/src/process/program_loader/elf/load_elf.rs

```diff
@@ -328,15 +328,17 @@ fn map_segment_vmo(
     }
 
     let perms = parse_segment_perm(program_header.flags);
-    let mut vm_map_options = root_vmar
-        .new_map(segment_size, perms)?
-        .vmo(segment_vmo)
-        .vmo_offset(segment_offset)
-        .vmo_limit(segment_offset + segment_size)
-        .can_overwrite(true);
     let offset = base_addr + (program_header.virtual_addr as Vaddr).align_down(PAGE_SIZE);
-    vm_map_options = vm_map_options.offset(offset).handle_page_faults_around();
-    let map_addr = vm_map_options.build()?;
+    if segment_size != 0 {
+        let mut vm_map_options = root_vmar
+            .new_map(segment_size, perms)?
+            .vmo(segment_vmo)
+            .vmo_offset(segment_offset)
+            .vmo_limit(segment_offset + segment_size)
+            .can_overwrite(true);
+        vm_map_options = vm_map_options.offset(offset).handle_page_faults_around();
+        let map_addr = vm_map_options.build()?;
```

Comment:
The variable `map_addr` is defined but never used, so there's no need to include `let map_addr` in this context. You can simply call `vm_map_options.build()?;` directly instead.